util/encoding/encoding.go

```diff
@@ -503,6 +511,20 @@ func DecodeIfNull(b []byte) ([]byte, bool) {
 	return b, false
 }
 
+// DecodeIfNotNull decodes a not-NULL value from the input buffer. If the input
+// buffer contains a not-NULL marker at the start of the buffer then it is
+// removed from the buffer and true is returned for the second
+// result. Otherwise, the buffer is returned unchanged and false is returned
+// for the second result. Note that the not-NULL marker is identical to the
+// empty string encoding, so do not use this routine where it is necessary to
+// distinguish not-NULL from the empty string.
+func DecodeIfNotNull(b []byte) ([]byte, bool) {
+	if len(b) >= 2 && b[0] == 0x00 && b[1] == 0x01 {
```

Comment:
This would have to be `len(b) >= 2 && bytes.Equal(b[:2], encodedNotNull)`. I'd be willing to bet the function call overhead would dominate any speedup from the asm `bytes.Equal` as we're only comparing 2 bytes. Alternately, I could use `bytes.HasPrefix` for legibility (which internally calls `bytes.Equal`). I doubt the perf difference will matter here.